Gemfile

```diff
@@ -1,16 +1,49 @@
 require 'json'
-require 'open-uri'
-
-versions = JSON.parse(open('https://pages.github.com/versions.json').read)
+require 'net/http'
 
 source 'https://rubygems.org'
 
-# The reason to add a ruby version at .ruby-version and here is to prevent
-# the CI server to push to Github if the versions mismatch.
-# see <https://circleci.com/docs/unrecognized-ruby-version/>
-ruby versions['ruby'] if ENV['CI']
+begin
+  uri      = URI('https://pages.github.com/versions.json')
+  versions = JSON.parse(Net::HTTP.get(uri))
+
+  # We need to be sure jekyll gets built in CI using the same version
+  # than Github does.
+  gem 'github-pages', versions['github-pages'], group: :jekyll_plugins
+
+  # The reason to add a ruby version at .ruby-version and here is to prevent
+  # the CI server to push to Github if the versions mismatch.
+  # see <https://circleci.com/docs/unrecognized-ruby-version/>
+  ruby versions['ruby'] if ENV['CI']
+
+# If for some reason the production versions check fails, we need to provide
+# a fallback scenario to the user.
+rescue Exception => exception
```

Comment:
I thought it was [better](http://stackoverflow.com/questions/10048173/) to use the default type rather than specifying `Exception`… ?